core/transaction.go

```diff
@@ -186,64 +228,36 @@ func (l *L1HandlerTransaction) signature() []*felt.Felt {
 	return make([]*felt.Felt, 0)
 }
 
-type DeployAccountTransaction struct {
-	Hash *felt.Felt
-	// Additional information given by the sender, used to validate the transaction.
-	Signature []*felt.Felt
-	// todo: add more
-}
-
-func (d *DeployAccountTransaction) hash() *felt.Felt {
-	return d.Hash
-}
-
-func (d *DeployAccountTransaction) signature() []*felt.Felt {
-	return d.Signature
-}
-
-func TransactionHash(transaction Transaction, network utils.Network) (*felt.Felt, error) {
+func transactionHash(transaction Transaction, n utils.Network) (*felt.Felt, error) {
 	switch t := transaction.(type) {
 	case *DeclareTransaction:
-		return declareTransactionHash(t, network)
+		return declareTransactionHash(t, n)
 	case *InvokeTransaction:
-		return invokeTransactionHash(t, network)
+		return invokeTransactionHash(t, n)
 	case *DeployTransaction:
-		return deployTransactionHash(t, network)
+		// deploy transactions are deprecated after re-genesis therefore we don't verify
+		// transaction hash
+		return nil, nil
 	case *L1HandlerTransaction:
-		return l1HandlerTransactionHash(t, network)
+		return l1HandlerTransactionHash(t, n)
 	case *DeployAccountTransaction:
-		return deployAccountTransactionHash(t, network)
+		return deployAccountTransactionHash(t, n)
 	default:
 		return nil, ErrUnknownTransaction
 	}
 }
 
-func deployTransactionHash(d *DeployTransaction, network utils.Network) (*felt.Felt, error) {
-	snKeccakConstructor, err := crypto.StarknetKeccak([]byte("constructor"))
-	if err != nil {
-		return nil, err
-	}
-	return crypto.PedersenArray(
-		new(felt.Felt).SetBytes([]byte("deploy")),
-		d.Version,
-		d.ContractAddress,
-		snKeccakConstructor,
-		crypto.PedersenArray(d.ConstructorCallData...),
-		new(felt.Felt),
-		network.ChainId(),
-	), nil
-}
-
-func invokeTransactionHash(i *InvokeTransaction, network utils.Network) (*felt.Felt, error) {
-	invokeFelt := new(felt.Felt).SetBytes([]byte("invoke"))
+var (
+	invokeFelt        = new(felt.Felt).SetBytes([]byte("invoke"))
+	declareFelt       = new(felt.Felt).SetBytes([]byte("declare"))
+	l1HandlerFelt     = new(felt.Felt).SetBytes([]byte("l1_handler"))
+	deployAccountFelt = new(felt.Felt).SetBytes([]byte("deploy_account"))
+)
+
+func invokeTransactionHash(i *InvokeTransaction, n utils.Network) (*felt.Felt, error) {
 	if i.Version.IsZero() {
-		return crypto.PedersenArray(
-			invokeFelt,
-			i.ContractAddress,
-			i.EntryPointSelector,
-			crypto.PedersenArray(i.CallData...),
-			network.ChainId(),
-		), nil
+		// Due to inconsistencies in version 0 hash calculation we don't verify the hash
+		return nil, nil
```

Comment:
returning `nil` Felt seems dangerous, can we just return `i.Hash` so that we dont have to check for `nil` in `verifyTransactionHash`?

Reply:
ok sure